skia/RSkSVGContainer.cpp

```diff
@@ -0,0 +1,190 @@
+
+  if(!childComponent || !childComponent.get()) {
+    RNS_LOG_ERROR("Invalid child component received");
+    return;
+  }
+
+  auto node=getRSKSvgNodeFromComponent(childComponent);
+  if(node) {
+    auto sksvgNode =sk_sp<SkSVGNode>(node);
+    RNS_LOG_MODE_FOR_CONTAINER("Child Node type  : "<<(int)node->tag());
+    if(!node->nodeName.empty()) {
+      RNS_LOG_MODE_FOR_CONTAINER("Child Node has ID : "<<node->nodeName);
+      nodeIDMapper_.set(SkString(node->nodeName),sksvgNode);
+    }
+
+    // If Child Node is of type of container , Merge it's defMap with it's Parent
+    auto nodeContainer=dynamic_cast<RSkSVGContainer *>(node);
+    if(nodeContainer) {
+      auto mergeFn= [&](SkString keyName,sk_sp<SkSVGNode> *value){
+        nodeIDMapper_.set(keyName,*value);
+      };
+      nodeContainer->nodeIDMapper_.foreach(mergeFn);
+    }
+
+    // Insert child RSkSVGNode in to List
+    size_t indexAt = std::min((size_t)index, childRSkNodeList_.size());
+    RNS_LOG_MODE_FOR_CONTAINER("Insert Child at index : " << indexAt );
+    childRSkNodeList_.insert(childRSkNodeList_.begin() + indexAt, sksvgNode);
+
+    //If this parent is SVGView , set itself as Root
+    if(this->tag() == SkSVGTag::kSvg) {
+      node->setRoot(this);
+    }
+  }
+
+  RNS_LOG_MODE_FOR_CONTAINER("RSK SVG Def Map Size for component : "<< this->getComponentData().componentName << " : "<<nodeIDMapper_.count());
+  RNS_LOG_MODE_FOR_CONTAINER("RSK SVG container Size for the Parent : "<<childRSkNodeList_.size());
+}
+
+void RSkSVGContainer::unmountChildComponent(
+    std::shared_ptr<RSkComponent> oldChildComponent,
+    const int index) {
+  RNS_LOG_MODE_FOR_CONTAINER("Parent :"<<getComponentData().componentName<<
+                    "recieved unmount for child :" << oldChildComponent->getComponentData().componentName<<
+                    "@ index : "<<index);
+  if(index >= childRSkNodeList_.size()) {
+    RNS_LOG_ERROR("Invalid index passed for remove");
+    return;
+  }
+  // If the respective node has entry In Def ID Mapper , remove it from all teh containers
+  auto node=static_cast<RSkSVGNode *>(childRSkNodeList_[index].get());
+  if(node && (!node->nodeName.empty())) {
+    auto node=dynamic_cast<RSkSVGContainer *>(rootNode_);
+    if(node) {
+     node->removeDefEntry(node->nodeName);
+    }
+  }
+  childRSkNodeList_.erase(childRSkNodeList_.begin() + index);
+}
+
+void RSkSVGContainer::OnPaint(SkCanvas *canvas) {
+  RNS_LOG_DEBUG(" Paint for Svg Elements handled via RNSSVG element ");
+}
+
+bool RSkSVGContainer::hasChildren() const {
+  return !childRSkNodeList_.empty();
+}
+
+void RSkSVGContainer::onRender(const SkSVGRenderContext& ctx) const {
+  for (int i = 0; i < childRSkNodeList_.size(); ++i) {
+    RNS_LOG_MODE_FOR_CONTAINER(" Child Tag : "<<(int)childRSkNodeList_[i]->tag());
+    childRSkNodeList_[i]->render(ctx);
+  }
+}
+
+void RSkSVGContainer::setRoot(RSkSVGNode * rootNode) {
+
```

Comment:
Ok. It is set Svg container as root to it's child. Which is to enable reach out to it's Parnet container to retrieve info like width of the container. It will needed for element liek TSpan for creating paragraph layout.